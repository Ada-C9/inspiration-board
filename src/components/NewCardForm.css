/* The Modal (background) */

.new-card-form {
  width: 50%;
  margin: auto;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
  font-size: 2em;
}

.new-card-form__form {
  font-size: 1.25em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr 2fr;
  grid-auto-flow: row;
  grid-gap: .8em;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  font-size: 1em;
  grid-column: controls;
  grid-row: auto;
}

.new-card-form__form-button {
  background-color: #F4FF81;
  border: 1.5px solid black;
  border-radius: 5px;
  font-size: 1em;
}

input {
  border: 1.5px solid black;
  border-radius: 5px;
  text-align: center;
  animation: pulse 5s infinite;
}

input:hover {
  background-color: #F4FF81;
}

select, input[type="color" i][list] {
  border: 1.5px solid black;
}

select:hover {
  background-color: #F4FF81;
}

@keyframes pulse {
  0%   {color: #FF0000;}
  25%  {color: #FF00FF;}
  50%  {color: #32CD32;}
  100% {color: #FF1493;}
}
