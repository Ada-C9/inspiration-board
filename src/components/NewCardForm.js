import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './NewCardForm.css';


class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: '',
    };
  }

  onFieldChange = (event) => {
    const fieldName = event.target.name;
    const fieldValue = event.target.value;
    const updateState = {};
    updateState[fieldName] = fieldValue;
    this.setState(updateState);
  }

  clearForm = () => {
    this.setState({
      text: '',
      emoji: ''
    });
  }
  onFormSubmit = (event) => {
    event.preventDefault();
    this.props.addCard(this.state)
    this.clearForm()
  }


  render() {
    return (
      <form onSubmit={this.onFormSubmit} className="new-card-form">
        <div className="new-card-form__form">
          <label className="new-card-form__form-label" htmlFor="text">Text: </label>
          <input
            className="new-card-form__form-textarea"
           name="text"
           value={this.state.text}
           onChange={this.onFieldChange}
           type="text" id="text_form"/>
        </div>
        <div className="new-card-form__form">
          <label htmlFor="emoji">Emoji: </label>
          <input
          name="emoji"
          value={this.state.emoji}
          onChange={this.onFieldChange}
          type="text"
          className="new-card-form__form-textarea" id="emoji_form"/>
        </div>

        <input
          className="new-card-form__form-button"
          type="submit"
          value="Inspire"
        />
      </form>
    );
  }
}

NewCardForm.propTypes = {
  addCard: PropTypes.func,
}

export default NewCardForm;
