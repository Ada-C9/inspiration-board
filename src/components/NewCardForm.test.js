import React from 'react';
import NewCardForm from './NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow( <NewCardForm addCardCallback={() => {} } />);
    expect(wrapper).toMatchSnapshot();
  });

  test('when a user enters text in the text field the field is updated', () => {

    const wrapper = shallow( <NewCardForm
      addCard={() => {} } />);

    let nameField = wrapper.find('input[name="text"]');

    nameField.simulate('change', {
      target: {
        value: 'testing out a positive note',
        name: 'text',
      },
    });

    wrapper.update();

    nameField = wrapper.find('input[name="text"]');
  
    expect(nameField.getElement().props.value).toEqual('testing out a positive note');
  });

  test('when a user enters text in the emoji field the field is updated', () => {

    const wrapper = shallow( <NewCardForm
      addCard={() => {} } />);

    let nameField = wrapper.find('input[name="emoji"]');

    nameField.simulate('change', {
      target: {
        value: 'heart_eyes',
        name: 'emoji',
      },
    });

    wrapper.update();

    nameField = wrapper.find('input[name="emoji"]');

    expect(nameField.getElement().props.value).toEqual('heart_eyes');
  });

});
