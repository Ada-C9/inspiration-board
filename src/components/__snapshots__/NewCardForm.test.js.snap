// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
<form
  className="new-card-form"
  onSubmit={[Function]}
>
  <div
    className="new-card-form__form"
  >
    <label
      className="new-card-form__form-label"
      htmlFor="text"
    >
      Text: 
    </label>
    <input
      className="new-card-form__form-textarea"
      id="text_form"
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="new-card-form__form"
  >
    <label
      htmlFor="emoji"
    >
      Emoji: 
    </label>
    <input
      className="new-card-form__form-textarea"
      id="emoji_form"
      name="emoji"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <input
    className="new-card-form__form-button"
    type="submit"
    value="Inspire"
  />
</form>
`;
